Documentation/ColumnsConfig/CommonProperties/ItemsProcFunc.rst

```diff
@@ -9,62 +9,86 @@ itemsProcFunc
-   :Types: :ref:`check <columns-check>`, :ref:`flex <columns-flex>`,
-      :ref:`radio <columns-radio>`
+   :Types: :ref:`check <columns-check>`, :ref:`select <columns-select>`, :ref:`radio <columns-radio>`
 
-   PHP function which is called to fill / manipulate the array with elements.
+   PHP method which is called to fill or manipulate the items array.
 
-   [classname]->[methodname]
+   VENDOR\\Extension\\UserFunction\\YourClass->yourMethod
 
-   The function/method will have an array of parameters passed to it, the items-array is passed by reference
-   in the key 'items'. By modifying the array of items, you alter the list of items. A method may throw an
+   The provided method will have an array of parameters passed to it. The items array is passed by reference
+   in the key :php:`items`. By modifying the array of items, you alter the list of items. A method may throw an
    exception which will be displayed as a proper error message to the user.
 
-   **Example:**
-
-   The configuration for the field `category_field` of the table `tt_content` looks like this:
-
-   .. code-block:: php
-
-       'type' => 'select',
-       'renderType' => 'selectSingle',
-       'itemsProcFunc' => 'TYPO3\CMS\Core\Category\CategoryRegistry->getCategoryFieldsForTable',
-       'itemsProcConfig' => [
-          'table' => 'tt_content'
-       ],
-       'minitems' => 0,
-       'maxitems' => 1,
-       'size' => 1
-
-   The referenced `itemsProcFunc` should populate the items by filling :php:`$configuration['items']``:
-
-   .. code-block:: php
-
-       /**
-        * Returns a list of category fields for a given table for populating selector "category_field"
-        * in tt_content table (called as itemsProcFunc).
-        *
-        * @param array $configuration Current field configuration
-        * @throws \UnexpectedValueException
-        * @internal
-        */
-       public function getCategoryFieldsForTable(array &$configuration)
-       {
-          $table = $configuration['config']['itemsProcConfig']['table'] ?? '';
-          // Return early if no table is defined
-          if (empty($table)) {
-             throw new \UnexpectedValueException('No table is given.', 1381823570);
-          }
-          // Loop on all registries and find entries for the correct table
-          foreach ($this->registry as $tableName => $fields) {
-             if ($tableName === $table) {
-                foreach ($fields as $fieldName => $options) {
-                   $fieldLabel = $this->getLanguageService()->sL($GLOBALS['TCA'][$tableName]['columns'][$fieldName]['label']);
-                   $configuration['items'][] = [$fieldLabel, $fieldName];
-                }
-             }
-          }
-       }
-
-   Further examples on different config type's can be found in the extension `styleguide`.
+   Passed parameters
+     * :php:`items` (passed by reference)
+     * :php:`config` (TCA config of the field)
+     * :php:`TSconfig`
```

Comment:
```suggestion
     * :php:`TSconfig` (currently active page TSconfig)
```

Reply:
Actually it's not the page tsconfig but this: https://docs.typo3.org/m/typo3/reference-tsconfig/master/en-us/PageTsconfig/TceForm.html?highlight=itemsprocfunc#itemsprocfunc

Do you know how to interlink to this?
It starts with `<t3tsconfig:` but I don't know what comes after. Any idea?

Reply:
You can actually see it in your link: `itemsprocfunc`. :-)

All IDs/anchors are actually used for linking.

See https://docs.typo3.org/m/typo3/docs-how-to-document/master/en-us/WritingReST/Hyperlinks.html#from-one-manual-to-another

Reply:
Thank you. It seems most headers have a reference to them like `.. _itemsProcFunc:`. This is missing for this exact option.